celery/worker/consumer/consumer.py

```diff
@@ -442,10 +452,28 @@ def _error_handler(exc, interval, next_step=CONNECTION_RETRY_STEP):
                 max_retries=self.app.conf.broker_connection_max_retries)
             error(CONNECTION_ERROR, conn.as_uri(), exc, next_step)
 
-        # remember that the connection is lazy, it won't establish
+        # Remember that the connection is lazy, it won't establish
         # until needed.
-        if not self.app.conf.broker_connection_retry:
-            # retry disabled, just call connect directly.
+        # If broker_connection_retry_on_startup is not set, revert to broker_connection_retry
+        # to determine whether connection retries are disabled.
+
+        # TODO: Rely only on broker_connection_retry_on_startup to determine whether connection retries are disabled.
+        #       We will make the switch in Celery 6.0.
+        deprecated_no_retry_on_startup = (self.app.conf.broker_connection_retry_on_startup is None
+                                          and not self.app.conf.broker_connection_retry)
+        if (
+            deprecated_no_retry_on_startup
+            or not self.app.conf.broker_connection_retry_on_startup
+        ):
+            if deprecated_no_retry_on_startup:
+                warnings.warn(
```

Comment:
@naomielst Please write a test which checks that we are warning about the pending deprecation here.

Reply:
Done